yt_idefix/data_structures.py

```diff
@@ -427,15 +427,175 @@ def _get_time(self):
 
     def _set_code_unit_attributes(self):
         """Conversion between physical units and code units."""
-        length_unit = self.parameters.get("unit_length", 1.0)
-        velocity_unit = self.parameters.get("unit_velocity", 1.0)
-        density_unit = self.parameters.get("unit_density", 1.0)
 
-        self.length_unit = self.quan(length_unit, "cm")
-        self.velocity_unit = self.quan(velocity_unit, "cm/s")
-        density_unit = self.quan(density_unit, "g/cm**3")
+        # Overriding units are preferred to PLUTO's base units
+        pluto_units = {
+            "length_unit": "cm",
+            "velocity_unit": "cm/s",
+            "density_unit": "g/cm**3",
+        }
+        spu = set(pluto_units)
+        if len(self.units_override) < 3:
+            if len(self.units_override):
+                ytLogger.info(
+                    "Less than 3 units were specified in units_override (%s). "
+                    "Need to rely on PLUTO's internal units to derive other units",
+                    len(self.units_override),
+                )
+
+            spu.difference_update(set(self.units_override))
```

Comment:
> note however that I don't think `self.units_override` should be updated internally. Because this attribute is user-facing, it should only reflect exactly what the user passed at load time, so I'd advise using a different intermediate variable here

I did considered that! But I also want to let users know from which three units the others are derived when they only pass 1 or 2 units, even though the additional units picked from Pluto definitions don't be overrided really. Maybe I should create another attribute?